pydantic_ai_slim/pydantic_ai/_parts_manager.py

```diff
@@ -67,20 +118,49 @@ def get_parts(self) -> list[ModelResponsePart]:
         """
         return [p for p in self._parts if not isinstance(p, ToolCallPartDelta)]
 
-    def handle_text_delta(
+    def handle_text_delta(  # noqa: C901
         self,
         *,
         vendor_part_id: VendorId | None,
         content: str,
         id: str | None = None,
-        thinking_tags: tuple[str, str] | None = None,
+        thinking_tags: ThinkingTags | None = None,
         ignore_leading_whitespace: bool = False,
-    ) -> ModelResponseStreamEvent | None:
+    ) -> Sequence[ModelResponseStreamEvent]:
         """Handle incoming text content, creating or updating a TextPart in the manager as appropriate.
 
-        When `vendor_part_id` is None, the latest part is updated if it exists and is a TextPart;
-        otherwise, a new TextPart is created. When a non-None ID is specified, the TextPart corresponding
-        to that vendor ID is either created or updated.
+        This function also handles what we'll call "loose thinking", which is the generation of
```

Comment:
I went Shakespeare on this docstring, let me know if it's too verbose. I thought it'd be good to document the limits of handling this in the code, could make it easier in the future to debug or adapt.